build/dart_rules.gni

```diff
@@ -0,0 +1,239 @@
+# Copyright 2015 The Chromium Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+# This file has rules for making Dart packages and Dart-based Mojo applications.
+# The entrypoint is the dart_pkg rule.
+
+import("//build/module_args/dart.gni")
+
+template("dart_pkg_helper") {
+  assert(defined(invoker.package_name))
+  package_name = invoker.package_name
+  pkg_directory = rebase_path("$root_gen_dir/dart-pkg")
+  package_root = rebase_path("$root_gen_dir/dart-pkg/packages")
+  stamp_file = "$root_gen_dir/dart-pkg/${package_name}.stamp"
+  entries_file = "$root_gen_dir/dart-pkg/${package_name}.entries"
+
+  assert(defined(invoker.sources) || defined(invoker.apps) ||
+         defined(invoker.libs) || defined(invoker.pkg_dir))
+
+  action(target_name) {
+    deps = []
+    if (defined(invoker.deps)) {
+      deps += invoker.deps
+    }
+
+    datadeps = []
+    if (defined(invoker.datadeps)) {
+      datadeps += invoker.datadeps
+    }
+
+    sdk_ext_directory = []
+    if (defined(invoker.sdk_ext_directory)) {
+      sdk_ext_directory += [ invoker.sdk_ext_directory ]
+    }
+
+    sdk_ext_files = []
+    if (defined(invoker.sdk_ext_files)) {
+      sdk_ext_files += invoker.sdk_ext_files
+    }
+
+    sdk_ext_mappings = []
+    if (defined(invoker.sdk_ext_mappings)) {
+      sdk_ext_mappings += invoker.sdk_ext_mappings
+    }
+
+    script = rebase_path("build/dart/dart_pkg.py", ".", "//")
```

Comment:
The only changes in this file are the removal of the mojo.gni import and the use of mojo_sdk_root to rebase various paths (now replaced by "//').